python_modules/libraries/dagster-k8s/dagster_k8s/client.py

```diff
@@ -563,13 +576,28 @@ def wait_for_pod(
                 self.sleeper(wait_time_between_attempts)
                 continue
 
-            if not pod.status.container_statuses:
-                self.logger("Waiting for pod container status to be set by kubernetes...")
+            if not pod.status.init_container_statuses and not pod.status.container_statuses:
+                self.logger(
+                    "Waiting for pod init_container or container status to be set by kubernetes..."
+                )
                 self.sleeper(wait_time_between_attempts)
                 continue
 
             # https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#containerstatus-v1-core
-            container_status = pod.status.container_statuses[0]
+            all_statuses = []
+            all_statuses.extend(pod.status.init_container_statuses or [])
+            all_statuses.extend(pod.status.container_statuses or [])
+
+            # Filter out ignored containers
+            all_statuses = [s for s in all_statuses if s.name not in ignore_containers]
```

Comment:
We could use set arithmetics, but this may be more readable. Let me know about your preference.

Reply:
this is fine by me